### PR TITLE
[WFLY-19589] Upgrade the MicroProfile REST Client to 4.0 and the RESTEasy implementation to 3.0.0.Final.

### PR DESCRIPTION
**Pull request by jamezp**
- https://issues.redhat.com/browse/WFLY-19589
- https://issues.redhat.com/browse/WFLY-19697

This also includes #18164

### CODE REVIEW

File: testsuite/integration/microprofile-tck/rest-client/src/test/java/org/wildfly/cleanup/WiremockCleanup.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+public class WiremockCleanup {
+
+    public static void main(String[] args) {
+        for (String arg : args) {
+            if (arg.isBlank()) {
+                continue;
+            }
+            System.out.printf("Cleaning up system property: %s=%s%n", arg, System.getProperty("javax.xml.transform.TransformerFactory"));
```
Comment (bstansberry) [inline testsuite/integration/microprofile-tck/rest-client/src/test/java/org/wildfly/cleanup/WiremockCleanup.java:18]: I feel lame asking for corrected logging from such an egregious hack, but...

It should be "System.getProperty(arg)"

:-D

Comment (jamezp) [reply testsuite/integration/microprofile-tck/rest-client/src/test/java/org/wildfly/cleanup/WiremockCleanup.java]: Doh, yes. Changed.

**Comment by bstansberry** [comment]
@jamezp This needs a rebase.

**Comment by jamezp** [comment]
> @jamezp This needs a rebase.

I had a feeling this was going to happen :) I can probably guess which PR caused the conflict :)

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @jamezp

**Pull request merged**